src/uucore/src/lib/mods/locale.rs

```diff
@@ -107,7 +107,68 @@ thread_local! {
     static LOCALIZER: OnceLock<Localizer> = const { OnceLock::new() };
 }
 
-/// Initialize localization with a specific locale and config
+/// Helper function to find the uucore locales directory from a utility's locales directory
+fn find_uucore_locales_dir(utility_locales_dir: &Path) -> Option<PathBuf> {
+    // Normalize the path to get absolute path
+    let normalized_dir = utility_locales_dir
+        .canonicalize()
+        .unwrap_or_else(|_| utility_locales_dir.to_path_buf());
+
+    // Walk up: locales -> printenv -> uu -> src
+    let uucore_locales = normalized_dir
+        .parent()? // printenv
+        .parent()? // uu
+        .parent()? // src
+        .join("uucore")
+        .join("locales");
+
+    // Only return if the directory actually exists
+    uucore_locales.exists().then_some(uucore_locales)
+}
+
+/// Create a bundle that combines common and utility-specific strings
+fn create_bundle(
+    locale: &LanguageIdentifier,
+    locales_dir: &Path,
+    util_name: &str,
+) -> Result<FluentBundle<FluentResource>, LocalizationError> {
+    let mut bundle = FluentBundle::new(vec![locale.clone()]);
+
+    // Disable Unicode directional isolate characters
+    bundle.set_use_isolating(false);
+
+    // Load common strings from uucore locales directory
+    if let Some(common_dir) = find_uucore_locales_dir(locales_dir) {
+        let common_locale_path = common_dir.join(format!("{locale}.ftl"));
+        if let Ok(common_ftl) = fs::read_to_string(&common_locale_path) {
+            if let Ok(common_resource) = FluentResource::try_new(common_ftl) {
+                bundle.add_resource_overriding(common_resource);
+            }
+        }
+    }
+
+    // Then, try to load utility-specific strings from the utility's locale directory
+    let util_locales_dir = get_locales_dir(util_name).ok();
+    if let Some(util_dir) = util_locales_dir {
+        let util_locale_path = util_dir.join(format!("{locale}.ftl"));
+        if let Ok(util_ftl) = fs::read_to_string(&util_locale_path) {
+            if let Ok(util_resource) = FluentResource::try_new(util_ftl) {
+                bundle.add_resource_overriding(util_resource);
+            }
+        }
+    }
```

Comment:
The code is in both cases the same and thus it would make sense to extract it into a function.